frontend/src/components/Bill/Bill.jsx

```diff
@@ -1,15 +1,25 @@
-import {PropTypes} from "prop-types";
-import {useState, useEffect} from "react";
+import { PropTypes } from "prop-types";
+import { useState, useEffect, useCallback } from "react";
 import { useParams } from "react-router-dom";
 
-const RelatedBills = ({related}) => {
+import './Bill.css';
+
+const MAPBOX_TOKEN = "sk.eyJ1IjoidHdibGFja3dlbGwiLCJhIjoiY2x0cWMzOWsxMDRtYjJrcXdmN2R1Zml3bSJ9.IABJE1MnM1zJeYkrGUGJ9g";
+const OPENSTATES_API_KEY = "2a046c1a-de61-4a68-8ad8-f6a3fd0c42f3";
```

Comment:
API token handling is _incredibly important_ so please read all of this.

1. don't put an API key in code
This repo is public, so everybody can view your API key. There are bots that scan GH for API keys.

2. don't use an API key on the frontend (if possible). I can see your tokens on the frontend:
![Screen Shot 2024-03-14 at 1 44 26 PM](https://github.com/sunrisemvmtnyc/legislation-tracker/assets/8884571/c79d309a-beb9-47e5-b62b-82ca97253718)

`MAPBOX_TOKEN` could be used, but you would [use a URL restriction](https://docs.mapbox.com/help/troubleshooting/how-to-use-mapbox-securely/#access-tokens)
`OPENSTATES_API_KEY` has no URL restrictions, so it should be used backend-only.

[Here is a good article](https://support.google.com/googleapi/answer/6310037?hl=en)

With those things in mind, I think you should:
1. [backend] Make an API endpoint on the backend that takes a user's zip code (or address?).
2. [backend] Make these two requests to mapbox & openstates.
3. [backend] Backend returns the resulting JSON for the user's legislator
4. [frontend] Frontend makes one API call `/your/new/api/endpoint?zip=<user-provided-zip>`

The user flow appears the same, but the API keys are used on the backend only.

In summation:
- ☑️ move 3rd-party API calls to backend
- ☑️ go to mapbox & openstates and reset both of your API keys

DM me if you want to chat about this.

Reply:
totally agreed - oversight by me last night that I forgot to mention and fix this morning. I'll try to follow the pattern used for backend keys for sure